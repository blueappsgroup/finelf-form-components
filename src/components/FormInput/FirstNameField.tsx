import React, { FC } from 'react'
import { Field } from 'formik'

import { TextFieldProps } from '../../types'
import BaseField from './base'
import { validateText } from './validate'

const FirstNameField: FC<TextFieldProps> = ({
  id,
  label = 'ImiÄ™',
  placeholder,
  name,
  required = false,
  validate = validateText,
  showError = true,
}) => (
  <Field
    id={id}
    label={label}
    placeholder={placeholder}
    component={BaseField}
    required={required}
    showError={showError}
    name={name}
    validate={validate(required)}
  />
)

FirstNameField.defaultProps = {
  name: 'first_name',
}

export default FirstNameField
