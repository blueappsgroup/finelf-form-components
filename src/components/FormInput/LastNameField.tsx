import React, { ReactElement } from 'react'
import { TextFieldProps } from '../../types'
import TextField from './TextField'

const LastNameField: (props: TextFieldProps) => ReactElement = ({
  id,
  label = 'Nazwisko',
  placeholder = 'Nazwisko',
  name = 'last_name',
  required = true,
  validate,
  showError = true,
}) => (
  <TextField
    id={id}
    label={label}
    placeholder={placeholder}
    required={required}
    showError={showError}
    name={name}
    validate={validate}
  />
)

export default LastNameField
