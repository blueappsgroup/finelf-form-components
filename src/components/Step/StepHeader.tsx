import styled from 'styled-components'
import { device } from '../../consts/sizes'

type StepHeaderProps = {
  activeStep?: boolean
  theme: {
    stepHeaderActiveColor: string
    stepHeaderInActiveColor: string
    stepHeaderActiveBorderColor: string
    stepHeaderInActiveBorderColor: string
  }
}

const StepHeader = styled.div<StepHeaderProps>`
  display: flex;
  justify-content: center;
  flex-grow: 1;
  font-size: 16px;
  letter-spacing: 0px;
  font-weight: ${(props: StepHeaderProps): number =>
    props.activeStep ? 800 : 400};
  color: ${(props: StepHeaderProps): string =>
    props.activeStep
      ? props.theme.stepHeaderActiveColor
      : props.theme.stepHeaderInActiveColor};
  border-bottom: ${(props: StepHeaderProps): string =>
    props.activeStep
      ? `4px solid ${props.theme.stepHeaderActiveBorderColor}`
      : `2px solid ${props.theme.stepHeaderInActiveBorderColor}`};
  margin: 0px 0px 30px;
  padding-bottom: 10px;

  &:first-of-type {
    margin-right: 15px;
  }

  &:last-child {
    margin-left: 15px;
  }

  @media ${device.mobileL} {
    &:first-of-type {
      margin-right: 15px;
    }

    &:last-child {
      margin-left: 15px;
    }
  }
`

export default StepHeader
