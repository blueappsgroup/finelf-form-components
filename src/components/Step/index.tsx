import React, {
  FC,
  useContext,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react'
import styled from 'styled-components'
import { useFormikContext } from 'formik'

import Button from '../Button'
import StepHeader from './StepHeader'
import { FormContext } from '../../utils'
import { device } from '../../consts/sizes'

const Wrapper = styled.div<{ visible: boolean }>`
  display: ${(props): string => (props.visible ? 'block' : 'none')};
`

type ButtonWrapperTypes = { isFirstStep: boolean }

// prettier-ignore
const ButtonsWrapper = styled.div<ButtonWrapperTypes>`
  display: flex;
  flex-direction: column;
  
  & button {
    flex-grow: 1;
  }

  @media ${device.mobileL} {
    flex-direction: row;

    & button {
      &:first-of-type {
        margin-right: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '15px'};
      }

      &:last-child {
        margin-left: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '15px'};
      }
    }
  }

  @media ${device.tablet} {
    & button {
      &:first-of-type {
        margin-right: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '30px'};
      }

      &:last-child {
        margin-left: ${(props: ButtonWrapperTypes): string => props.isFirstStep ? '0px' : '30px'};
      }
    }
  }
`

const StepHeaderWrapper = styled.div`
  display: flex;
`

type Props = {
  stepIndex: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: any
  onStepComplete?: () => void
}

const Step: FC<Props> = ({ children, stepIndex, onStepComplete }) => {
  const wrapperRef = useRef<any>(null)
  const {
    currentStep = 0,
    stepsLength = 1,
    stepsTitleList,
    prevStep,
    nextStep,
  } = useContext(FormContext)
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const { values, errors } = useFormikContext<any>()

  const mappedFields = useMemo(
    () =>
      (Array.isArray(children) ? children : [children]).reduce(
        (acc, item) => {
          if (item.props.name && !item.props.children) {
            acc[item.props.name] = true
          }
          if (
            item.props.name &&
            item.props.required &&
            !item.props.visibleCondition
          ) {
            acc.requiredFields[item.props.name] = true
          }
          if (item.props && item.props.name === 'agreements') {
            acc.requiredFields[item.props.name] = true
          }

          if (item.props.children) {
            const mappedChildrens = Array.isArray(item.props.children)
              ? item.props.children
              : [item.props.children]

            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            mappedChildrens.forEach((child: any) => {
              if (
                child.props.name &&
                child.props.required &&
                !child.props.visibleCondition
              ) {
                if (item.props.type === 'checkboxGroup') {
                  !acc.requiredFields[item.props.name] &&
                    (acc.requiredFields[item.props.name] = {})
                  acc.requiredFields[item.props.name][child.props.name] = true
                  return
                }

                acc.requiredFields[child.props.name] = true
              }
              if (child.props.name) {
                acc[child.props.name] = true
              }
            })
          }

          return acc
        },
        { requiredFields: {} }
      ),
    [children]
  )

  const [nextButtonDisabled, setNextButtonDisabled] = useState(
    JSON.stringify(mappedFields) !== JSON.stringify({})
  )
  const lastStepIndex = stepsLength ? stepsLength - 1 : 0

  useEffect(() => {
    let hasError

    Object.keys(mappedFields.requiredFields).some((key: string) => {
      if (!values[key] || values[key] === '' || errors[key]) {
        hasError = true
        return true
      }

      if (key === 'agreements' && errors[key]) {
        hasError = true
        return true
      }

      return false
    })

    if (!hasError) {
      Object.keys(mappedFields).some((key: string) => {
        if (errors[key]) {
          hasError = true
          return true
        }

        return false
      })
    }

    if (hasError) {
      setNextButtonDisabled(true)
    }

    if (!hasError) {
      setNextButtonDisabled(false)
    }
  }, [values, errors, mappedFields, children])

  const handleClick = () => {
    if (onStepComplete !== undefined) {
      onStepComplete()
    }

    if (nextStep !== undefined) {
      nextStep()
    }
  }

  useEffect(() => {
    if (stepIndex === currentStep) {
      wrapperRef.current !== null &&
        window.scrollTo(0, wrapperRef.current.offsetTop)
    }
  }, [currentStep, stepIndex])

  return (
    <Wrapper ref={wrapperRef} visible={stepIndex === currentStep}>
      <StepHeaderWrapper>
        {currentStep !== 0 && (
          <StepHeader>
            {stepsTitleList && stepsTitleList[currentStep - 1]}
          </StepHeader>
        )}
        {stepIndex === currentStep && (
          <StepHeader activeStep>
            {stepsTitleList && stepsTitleList[currentStep]}
          </StepHeader>
        )}
        {currentStep === 0 && stepsLength > 1 && (
          <StepHeader>
            {stepsTitleList && stepsTitleList[currentStep + 1]}
          </StepHeader>
        )}
      </StepHeaderWrapper>
      {stepIndex === currentStep && children}
      <ButtonsWrapper isFirstStep={currentStep === 0}>
        {currentStep !== 0 && (
          <Button
            type="button"
            variant="secondary"
            text="Cofnij"
            onClick={prevStep}
          />
        )}
        {currentStep !== lastStepIndex && (
          <Button
            disabled={nextButtonDisabled}
            type="button"
            text="Dalej"
            onClick={handleClick}
          />
        )}
        {currentStep === lastStepIndex && (
          <Button disabled={nextButtonDisabled} type="submit" text="WyÅ›lij" />
        )}
      </ButtonsWrapper>
    </Wrapper>
  )
}

export default Step
