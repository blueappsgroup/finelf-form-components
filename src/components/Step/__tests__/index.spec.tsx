import * as React from 'react'
import { render, act, fireEvent } from '@testing-library/react'
import FirstNameField from '../../FormInput/FirstNameField'
import Form from '../../Form'
import LastNameField from '../../FormInput/LastNameField'
import Step from '../index'
import StepHeader from '../StepHeader'

describe('base <Step />', () => {
  const onSubmit = jest.fn()
  const stepsLength = 2
  const stepsTitles = ['1. Podstawowe dane', '2. Szczegółowe dane']
  const setupWrapper = ({
    formId = 'testForm',
    ...rest
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  }): any => {
    const wrapper = render(
      <Form
        id={formId}
        onSubmit={onSubmit}
        stepsLength={stepsLength}
        stepsTitles={stepsTitles}
      >
        <Step stepIndex={0}>
          <FirstNameField name="fName" required />
        </Step>
        <Step stepIndex={1}>
          <LastNameField name="lName" required />
        </Step>
      </Form>
    )
    const { container } = wrapper
    const firstName = container.querySelector('[name="fName"]')

    return {
      firstName,
      ...wrapper,
    }
  }

  it('matches snapshot', () => {
    const wrapper = setupWrapper({})

    expect(StepHeader).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('buttons disabled if required fields are not filled', () => {
    const wrapper = setupWrapper({})
    const { container } = wrapper
    const button = container.querySelector('button[disabled]')

    expect(button).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('buttons disabled if fields are filled improperly', async () => {
    const wrapper = setupWrapper({})
    const { container, firstName } = wrapper

    await act(async () => {
      fireEvent.change(firstName, { target: { value: '123' } })
    })

    const button = container.querySelector('button[disabled]')

    expect(button).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('setNextButtonDisabled have been called after fill step fields', async () => {
    const wrapper = setupWrapper({})
    const { container, firstName } = wrapper

    await act(async () => {
      fireEvent.change(firstName, { target: { value: 'test' } })
    })

    expect(container.querySelector('button:not([disabled])')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
