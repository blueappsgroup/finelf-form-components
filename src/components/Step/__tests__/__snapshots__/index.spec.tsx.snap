// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base <Step /> buttons disabled if fields are filled improperly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-lmgQde jpYwxd"
        id="testForm"
      >
        <input
          class="sc-cxNIbT kMEpPa"
          form="[object Object]"
        />
        <div
          class="sc-ezzayL CbuIx"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE lzsiK"
            >
              1. Podstawowe dane
            </h3>
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-hKFyIo gtdXMP"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-iemXMA dtwGgT"
            >
              <input
                class="sc-pNWxx hoRJAk"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value="123"
              />
            </div>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
        <div
          class="sc-ezzayL cGVRGm"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-lmgQde jpYwxd"
      id="testForm"
    >
      <input
        class="sc-cxNIbT kMEpPa"
        form="[object Object]"
      />
      <div
        class="sc-ezzayL CbuIx"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE lzsiK"
          >
            1. Podstawowe dane
          </h3>
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-hKFyIo gtdXMP"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-iemXMA dtwGgT"
          >
            <input
              class="sc-pNWxx hoRJAk"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value="123"
            />
          </div>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
      <div
        class="sc-ezzayL cGVRGm"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "firstName": <input
    class="sc-pNWxx hoRJAk"
    label="Imię"
    name="fName"
    placeholder="Imię*"
    required=""
    value="123"
  />,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Step /> buttons disabled if required fields are not filled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-lmgQde jpYwxd"
        id="testForm"
      >
        <input
          class="sc-cxNIbT kMEpPa"
          form="[object Object]"
        />
        <div
          class="sc-ezzayL CbuIx"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE lzsiK"
            >
              1. Podstawowe dane
            </h3>
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-hKFyIo gtdXMP"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-iemXMA dtwGgT"
            >
              <input
                class="sc-pNWxx hoRJAk"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
        <div
          class="sc-ezzayL cGVRGm"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-lmgQde jpYwxd"
      id="testForm"
    >
      <input
        class="sc-cxNIbT kMEpPa"
        form="[object Object]"
      />
      <div
        class="sc-ezzayL CbuIx"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE lzsiK"
          >
            1. Podstawowe dane
          </h3>
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-hKFyIo gtdXMP"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-iemXMA dtwGgT"
          >
            <input
              class="sc-pNWxx hoRJAk"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value=""
            />
          </div>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
      <div
        class="sc-ezzayL cGVRGm"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "firstName": <input
    class="sc-pNWxx hoRJAk"
    label="Imię"
    name="fName"
    placeholder="Imię*"
    required=""
    value=""
  />,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Step /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-lmgQde jpYwxd"
        id="testForm"
      >
        <input
          class="sc-cxNIbT kMEpPa"
          form="[object Object]"
        />
        <div
          class="sc-ezzayL CbuIx"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE lzsiK"
            >
              1. Podstawowe dane
            </h3>
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-hKFyIo gtdXMP"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-iemXMA dtwGgT"
            >
              <input
                class="sc-pNWxx hoRJAk"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value=""
              />
            </div>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
        <div
          class="sc-ezzayL cGVRGm"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-lmgQde jpYwxd"
      id="testForm"
    >
      <input
        class="sc-cxNIbT kMEpPa"
        form="[object Object]"
      />
      <div
        class="sc-ezzayL CbuIx"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE lzsiK"
          >
            1. Podstawowe dane
          </h3>
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-hKFyIo gtdXMP"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-iemXMA dtwGgT"
          >
            <input
              class="sc-pNWxx hoRJAk"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value=""
            />
          </div>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
      <div
        class="sc-ezzayL cGVRGm"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "firstName": <input
    class="sc-pNWxx hoRJAk"
    label="Imię"
    name="fName"
    placeholder="Imię*"
    required=""
    value=""
  />,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`base <Step /> setNextButtonDisabled have been called after fill step fields 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
        class="sc-lmgQde jpYwxd"
        id="testForm"
      >
        <input
          class="sc-cxNIbT kMEpPa"
          form="[object Object]"
        />
        <div
          class="sc-ezzayL CbuIx"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE lzsiK"
            >
              1. Podstawowe dane
            </h3>
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-hKFyIo gtdXMP"
          >
            <label
              for="fName"
            >
              Imię*
            </label>
            <div
              class="sc-iemXMA dtwGgT"
            >
              <input
                class="sc-pNWxx hoRJAk"
                label="Imię"
                name="fName"
                placeholder="Imię*"
                required=""
                value="test"
              />
            </div>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
        <div
          class="sc-ezzayL cGVRGm"
        >
          <div
            class="sc-kLokih glHjhv"
          >
            <h3
              class="sc-giAqnE gFdsiU"
            >
              2. Szczegółowe dane
            </h3>
          </div>
          <div
            class="sc-bYwzba gA-DsVm"
          >
            <button
              class="sc-iJCRLp cruxNc"
              disabled=""
              type="button"
            >
              Dalej
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
      class="sc-lmgQde jpYwxd"
      id="testForm"
    >
      <input
        class="sc-cxNIbT kMEpPa"
        form="[object Object]"
      />
      <div
        class="sc-ezzayL CbuIx"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE lzsiK"
          >
            1. Podstawowe dane
          </h3>
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-hKFyIo gtdXMP"
        >
          <label
            for="fName"
          >
            Imię*
          </label>
          <div
            class="sc-iemXMA dtwGgT"
          >
            <input
              class="sc-pNWxx hoRJAk"
              label="Imię"
              name="fName"
              placeholder="Imię*"
              required=""
              value="test"
            />
          </div>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
      <div
        class="sc-ezzayL cGVRGm"
      >
        <div
          class="sc-kLokih glHjhv"
        >
          <h3
            class="sc-giAqnE gFdsiU"
          >
            2. Szczegółowe dane
          </h3>
        </div>
        <div
          class="sc-bYwzba gA-DsVm"
        >
          <button
            class="sc-iJCRLp cruxNc"
            disabled=""
            type="button"
          >
            Dalej
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "firstName": <input
    class="sc-pNWxx hoRJAk"
    label="Imię"
    name="fName"
    placeholder="Imię*"
    required=""
    value="test"
  />,
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
